applications/materialDataCreator/materialDataCreator.html

```diff
@@ -0,0 +1,585 @@
+                    var ch = this[i];
+                    switch (ch)
+                    {
+                        case "\\": ch = "\\\\"; break;
+                        //case "\'": ch = "\\'"; break;
+                        case "\"": ch = '\\"'; break;
+                        //case "\&": ch = "\\&"; break;
+                        case "\t": ch = "\\t"; break;
+                        case "\n": ch = "\\n"; break;
+                        case "\r": ch = "\\r"; break;
+                        case "\b": ch = "\\b"; break;
+                        case "\f": ch = "\\f"; break;
+                        case "\v": ch = "\\v"; break;
+                        default: break;
+                    }
+                    
+                    result += ch;
+                }
+                
+                return result;
+            };
+            
+            var channel = "materialDataCreator.ak.vircadia";
+
+
+			function copyToClipboard(data) {
+				var $temp = $("<input>");
+				$("body").append($temp);
+				$temp.val(data).select();
+				document.execCommand("copy");
+				$temp.remove();
+			}
+            
+            function demo(shape) {
+                var message = {
+                    "channel": channel,
+                    "action": "changeDemoShape",
+                    "shape": shape
+                };
+                
+                EventBridge.emitWebEvent(JSON.stringify(message));            
+            
+            }
+            
+            function exportAsJson() {
+                var message = {
+                    "channel": channel,
+                    "action": "exportMaterialAsJson"
+                };
+                
+                EventBridge.emitWebEvent(JSON.stringify(message));            
+            
+            }
+
+            function createMaterialEntity() {
+                var message = {
+                    "channel": channel,
+                    "action": "createMaterialEntity"
+                };
+                
+                EventBridge.emitWebEvent(JSON.stringify(message));            
+            
+            }
+            
+            function teleportToServerless() {
+                 var message = {
+                    "channel": channel,
+                    "action": "teleportToServerless"
+                };
+                
+                EventBridge.emitWebEvent(JSON.stringify(message));  
+            }
+            
+            function toggleDayNight() {
+                 var message = {
+                    "channel": channel,
+                    "action": "toggleDayNight"
+                };
+                
+                EventBridge.emitWebEvent(JSON.stringify(message));  
+            }
+            
+        </script>
+    </head>
+    <body>
+        <form name='materialEditor' id='materialEditor'>
+        
+        <table style= 'width: 100%;'><tr><td style= 'width: 90%;'>

```

Comment:
Is there a reason you use 90% width here?
I assume that is why the app doesn't extend to the whole window.
![1Auswahl_849](https://user-images.githubusercontent.com/11144627/110458086-a59a4780-80cb-11eb-9845-d23f05654b5e.png)

Reply:
Yes, this is intentional. 
This is for the **sidewalk**... 

![image](https://user-images.githubusercontent.com/60075796/110570440-e3c35580-8123-11eb-98be-ea5125e11c6d.png)

...which is there for **HMD**. 

This is to have a dedicated space where you can grip to scroll the page up and down.
Otherwise manipulating it by the form's background is a pain since it constantly disturbs the different controls. 
And the scollbar is also a pain to use in HMD.

This is nothing standard. It's an innovation of mine specifically for HMD usage.

They wanted this to behave as a tablet, but the reality is more that we're manipulating the UI with a 16" stick, which makes the notion of comfort to be revisited. 
_(Eventually, in the future we will be able to use our hands with more precision and those kinds of things won't be necessary.)_